lib/Dialect/FIRRTL/Transforms/PrefixModules.cpp

```diff
@@ -128,8 +128,42 @@ void PrefixModulesPass::renameModuleBody(std::string prefix, FModuleOp module) {
   // changed yet.
   if (!prefix.empty())
     anythingChanged = true;
+  StringAttr thisMod = module.getNameAttr();
+  // A predicate to check if an annotation can be removed. If there is a
+  // refernece to a NLA, the NLA should either contain this module in its path,
+  // if its an InstanceOp. Else, it must exist at the leaf of the NLA. Otherwise
+  // the NLA reference can be removed, since its a spurious annotation, result
+  // of cloning the original module.
+  auto canRemoveAnno = [&](Annotation anno, Operation *op) -> bool {
+    auto nla = anno.getMember("circt.nonlocal");
+    if (!nla)
+      return false;
+    auto nlaName = nla.cast<FlatSymbolRefAttr>().getValue();
+    auto nlaIter = nlaMap.find(nlaName);
+    if (nlaIter == nlaMap.end()) {
+      op->emitError("cannot find NonLocalAnchor :" + nlaName);
+      return true;
+    }
 
+    auto nlaOp = dyn_cast<NonLocalAnchor>(nlaIter->second);
+    bool isValid = false;
+    if (isa<InstanceOp>(op))
+      isValid = nlaOp.hasModule(thisMod);
+    else
+      isValid = nlaOp.leafMod() == thisMod;
+    return !isValid;
+  };
+  // Remove spurious NLA references from the module ports and the module itself.
```

Comment:
shouldn't spurious be nla references be caught by verification?  Maybe I'm misunderstanding "spurious"

Reply:
As explained above the spurious NLA reference occurs because a module may be duplicated based on one of its instance paths. So, only some of the original `circt.nonlocal` NLA references are valid now. We donot remove the spurious NLAs during module duplication. It is removed later during traversal.